service/storage_proxy.cc

```diff
@@ -3296,6 +3308,25 @@ storage_proxy::create_write_response_handler(const std::tuple<lw_shared_ptr<paxo
                     inet_address_vector_topology_change(), inet_address_vector_topology_change(), std::move(tr_state), get_stats(), std::move(permit), std::monostate(), is_cancellable::no);
 }
 
+// After sending the write to replicas(targets), the replicas might generate send view updates. If the number of view updates
+// that the replica is already sending is too high, the replica should not send any more view updates, and consequently,
+// we should not send the write to the replica.
+template<typename Range>
+bool storage_proxy::can_update_views(const Range& targets, const schema_ptr& s, db::write_type type, tracing::trace_state_ptr tr_state) const {
+    static constexpr double mv_admission_control_threshold = 0.8;
+    if (type == db::write_type::VIEW) {
+        return true;
+    }
+    if (!_db.local().find_column_family(s).views().empty() && targets.begin() != targets.end()) {
```

Comment:
This is also unfortunate. We have to do a hash lookup only to find that there are no views. What could have been three instructions for an early return turns out to be many.

Again the fix isn't simple. We can't adjust the schema class since it's supposed to be immutable (except for true schema changes) and adding a view doesn't change the base table. I think we need to add a wrapper for the table+views and look it up instead of the schema, at the top level. /cc @tgrabiec

Reply:
The targets.empty() check is implicit in the for loop.

Reply:
We now have `s->table()` so we could avoid the lookup. There was controversy around it, since table is not part of the schema. But `s->views()` seems untroversial since we stay in the schema domain.

Reply:
I used the `s->table()` in the rebase for now.